File: Games/NoughtsAndCrosses/src/main/kotlin/jokatu/game/games/noughtsandcrosses/game/NoughtsAndCrossesGameFactory.java
package jokatu.game.games.noughtsandcrosses.game;

import jokatu.game.AbstractGameFactory;
import jokatu.game.GameID;
import org.jetbrains.annotations.NotNull;
import org.springframework.stereotype.Component;

@Component
public class NoughtsAndCrossesGameFactory extends AbstractGameFactory {

	@NotNull
	@Override
	protected NoughtsAndCrossesGame produce(@NotNull GameID gameID, @NotNull String creatorName) {
		return new NoughtsAndCrossesGame(gameID);
	}
}

File: Games/NoughtsAndCrosses/src/main/kotlin/jokatu/game/games/noughtsandcrosses/game/NoughtsAndCrossesGameFactory.kt
package jokatu.game.games.noughtsandcrosses.game

import jokatu.game.AbstractGameFactory
import jokatu.game.GameID
import org.springframework.stereotype.Component

@Component
class NoughtsAndCrossesGameFactory : AbstractGameFactory<NoughtsAndCrossesGame>() {

	override fun produce(gameID: GameID, creatorName: String): NoughtsAndCrossesGame {
		return NoughtsAndCrossesGame(gameID)
	}
}
